### PR TITLE
[User] Handle login page and logout

### PR DESCRIPTION
**Pull request by truongha-1240**
## Related Tickets
- [#47487](https://edu-redmine.sun-asterisk.vn/issues/47487)

## WHAT (optional)
- Handle login page for user
- Add logout feature

## Evidence (Screenshot or Video)
image
image
image

### CODE REVIEW

**Comment by truongha-1240** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -25,6 +25,22 @@ class User < ApplicationRecord
...
   has_secure_password
 
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false if digest.nil?
```
Comment (vanvtt-0952) [inline app/models/user.rb]:
```suggestion
    return false unless digest
```

**Comment by truongha-1240** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+      redirect_back_or root_url
+    else
+      message = t "sessions.create.no_activation"
+      flash[:warning] = message
+      redirect_to login_url
+    end
+  end
+
+  def check_authenticated
+    return true if @user&.authenticate params[:session][:password]
```
Comment (vanvtt-0952) [inline app/controllers/sessions_controller.rb:31]:
```suggestion
    return true if @user.authenticate params.dig(:session, :password)
```

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+
+    flash.now[:danger] = t "sessions.create.wrong_pass"
+    render :new
+  end
+
+  def find_user_by_email
+    @user = User.find_by email: params[:session][:email].downcase
```
Comment (vanvtt-0952) [inline app/controllers/sessions_controller.rb:38]:
```suggestion
    @user = User.find_by email: params.dig(:session, :email)&.downcase
```

**Pull request merged**